llmos-ai/llmos
# Add mirror & global system registry config

**Description:**
- Add mirror & global system registry config support.
- Bump go mod.

#### Testing Steps:
Case 1. Install LLMOS using mirror registry config:
  - Run`./llmos bootstrap --debug --mirror=cn --cluster-init`
  - Verify the `system-default-registry` is set to `registry.cn-hangzhou.aliyuncs.com` at `/etc/rancher/k3s/config.yaml.d/40-llmos.yaml`
  - Check the `/var/lib/llmos/charts/llmos-operator-config.yaml` config file and the `global.imageRegistry` will be set to:
  ```
  apiVersion: helm.cattle.io/v1
  kind: HelmChartConfig
  metadata:
    creationTimestamp: null
    name: llmos-operator
    namespace: llmos-system
  spec:
    valuesContent: |
      global:
        imageRegistry: llmos-ai-cn-beijing.cr.volces.com
  ```

Case 2. Set mirror and globalSystemRegistry config via `/etc/llmos/config.yaml` file
  ```
  role: cluster-init
  mirror: cn
  llmosInstallerImage: .. # using custom installer image if needed
  globalSystemImageRegistry: testing.registry.com:5000
  ```
  - Bootstrap the llmos `./llmos bootstrap --debug`
  - Verify that the globalImageRegistry is set to `testing.registry.com:5000` on both `/var/lib/llmos/bootstrapmanifests/llmos-prepost.yaml` and `/var/lib/llmos/charts/llmos-operator-config.yaml` config files.
